pip/server/server_test.go

```diff
@@ -0,0 +1,180 @@
+package server
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewServer(t *testing.T) {
+	s := NewServer(
+		WithNetwork("tcp4"),
+		WithAddress("127.0.0.1:5555"),
+	)
+
+	assert.Equal(t, s.opts.net, "tcp4")
+	assert.Equal(t, s.opts.addr, "127.0.0.1:5555")
+}
+
+func TestServerBind(t *testing.T) {
+	s := NewServer(
+		WithAddress("127.0.0.1:0"),
+	)
+
+	if assert.NoError(t, s.Bind()) {
+		assert.NotEqual(t, nil, s.ln)
+		assert.NoError(t, s.Stop())
+	}
+}
+
+func TestServerBindUnix(t *testing.T) {
+	dir, err := ioutil.TempDir("", "pip.server.test.")
+	if err != nil {
+		assert.FailNow(t, "failed to create temporary directory", "ioutil.TempDir error: %s", err)
+	}
+	defer func() {
+		if err = os.RemoveAll(dir); err != nil {
+			assert.Fail(t, "failed to cleanup temporary directory", "os.RemoveAll error: %s", err)
+		}
+	}()
+
+	tmp := path.Join(dir, "test.socket")
+
+	s := NewServer(
+		WithNetwork("unix"),
+		WithAddress(tmp),
+	)
+
+	if assert.NoError(t, s.Bind()) {
+		assert.NotEqual(t, nil, s.ln)
+		assert.NoError(t, s.Stop())
+	}
+}
+
+func TestServerBindTwice(t *testing.T) {
+	s := NewServer(
+		WithAddress("127.0.0.1:0"),
+	)
+
+	if assert.NoError(t, s.Bind()) {
+		assert.Equal(t, ErrBound, s.Bind())
+	}
+}
+
+func TestServerBindInvalidNetwork(t *testing.T) {
+	assert.NotEqual(t, nil, NewServer(WithNetwork("invalid")).Bind())
```

Comment:
consider splitting the code into several lines like other tests to make the tests style consistent and the code clearer